general/g.region/main.c

```diff
@@ -834,7 +834,7 @@ int main(int argc, char *argv[])
     /* grow by number of cells */
     if ((value = parm.grow->answer)) {
         update_file = true;
-        if (sscanf(value, "%i", &pix)) {
+        if (sscanf(value, "%i", &pix) == 1 ) {
```

Comment:
EOF "expands into a negative integral constant expression (generally, -1)" ([source](https://cplusplus.com/reference/cstdio/EOF/)), so `>=1` or, here, `==1` is required to get true for the "good" cases.

Reply:
What I mean is, when `parm.grow->answer` is not `NULL` (line 835), `*parm.grow->answer` is not '\0'. In this case, `sscanf()` cannot return `EOF`. [A quick search](https://www.ibm.com/docs/en/i/7.1?topic=functions-sscanf-read-data):
> The return value is EOF when **the end of the string** is encountered before anything is converted.

Reply:
[Here, it stays more general](https://en.cppreference.com/w/c/io/fscanf), noting that 

> 1-3) Number of receiving arguments successfully assigned (which may be zero in case a matching failure occurred before the first receiving argument was assigned), _or [EOF](https://en.cppreference.com/w/c/io) if input failure occurs before the first receiving argument was assigned._

What I understand from this is that there is some chance that another failure gets us an EOF, and it isn't only a question of strings or not. So handling correctly the return values _just because we can_ should be the correct thing to do.

Reply:
since this seems to be good practice, Ill keep it and changes similar to it in the PR.